Is it expected that the host.native subset fails on a fresh build (when the single file host isn't yet built)?

The NativeAOT libraries linker tests don't require the singlefilehost but the apphost but building the `host.native` fails if the single file host isn't yet built. Is that intentional? If the singlefilehost is a strong dependency to build `host.native` then we should update the Subset description to indicate that.

cc @agocke @sbomer 

```
C:\git\runtime2>build.cmd host.native -c Release
  Determining projects to restore...
  Tool 'coverlet.console' (version '3.1.2') was restored. Available commands: coverlet
  Tool 'dotnet-reportgenerator-globaltool' (version '5.0.2') was restored. Available commands: reportgenerator
  Tool 'microsoft.dotnet.xharness.cli' (version '1.0.0-prerelease.22566.1') was restored. Available commands: xharness
  Tool 'microsoft.visualstudio.slngen.tool' (version '8.5.17') was restored. Available commands: slngen
  Tool 'dotnet-format' (version '6.0.240501') was restored. Available commands: dotnet-format

  Restore was successful.
  All projects are up-to-date for restore.
  Determining projects to restore...
  All projects are up-to-date for restore.
  InjectResource -> C:\git\runtime2\artifacts\bin\coreclr\windows.x64.Release\InjectResource\InjectResource.dll
C:\git\runtime2\src\native\corehost\corehost.proj(48,5): error MSB3030: Could not copy the file "C:\git\runtime2\artifacts\bin\coreclr\windows.x64.Release\/corehost/si
nglefilehost.exe" because it was not found.
```

--- a/src/native/corehost/corehost.proj
+++ b/src/native/corehost/corehost.proj
@@ -54,6 +54,7 @@
   </Target>
 
   <Target Name="PrepareSingleFileHostWithEmbeddedDac"
+          Condition="Exists('$(SingleFileHostPath)')"
           DependsOnTargets="PrepareSingleFileHostWithEmbeddedDacPrereqs;PrepareSingleFileHostWithEmbeddedDacCore"
           AfterTargets="Build"/>
 

--- a/src/native/corehost/corehost.proj
+++ b/src/native/corehost/corehost.proj
@@ -44,7 +44,8 @@
             @(InjectResourceTool)"
     Outputs="$(SingleFileHostDestinationPath);$(SingleFileHostSymbolsDestinationPath)"
     DependsOnTargets="ResolveProjectReferences"
-    Condition="'$(RuntimeFlavor)' != 'Mono'">
+    Condition="'$(RuntimeFlavor)' != 'Mono' and
+               Exists('$(SingleFileHostPath)')">
     <Copy SourceFiles="$(SingleFileHostPath)" DestinationFiles="$(SingleFileHostDestinationPath)" />
     <Exec Condition="'$(TargetOS)' == 'windows'"
           Command="&quot;$(DotNetTool)&quot; exec @(InjectResourceTool) --bin &quot;$(DacPath)&quot; --image &quot;$(SingleFileHostDestinationPath)&quot; --name MINIDUMP_EMBEDDED_AUXILIARY_PROVIDER" />
@@ -54,7 +55,6 @@
   </Target>
 
   <Target Name="PrepareSingleFileHostWithEmbeddedDac"
-          Condition="Exists('$(SingleFileHostPath)')"
           DependsOnTargets="PrepareSingleFileHostWithEmbeddedDacPrereqs;PrepareSingleFileHostWithEmbeddedDacCore"
           AfterTargets="Build"/>
 

